### DISCUSSION TITLE
How to get scalp coordinates via python command line scripting?

### DISCUSSION THREAD

**Discussion by akelkar26** [simnibs/simnibs#199] [9. Results and analysis]
Hi,

Hope you are well. Thank you for the wonderful software. I have been able to do some novel analysis with it.

I have a few questions I need some insight over. 
My main goal is to find the the scalp coordinate from my tms simulation and the scalp coordinate from a tms optimization.

I was able to run the Simnibs TMS simulation by importing coordinates from Brainsight coordinates as seen from my affine transformation matsimnibs matrix. I also specified my didt value but didn't set `pos.distance` (it was default).

Questions:
1. As matsimnibs was shown when the simulation ran, does it mean matsimnibs was set? What is the `pos.distance` value the simulation ran on? 0mm or 4mm? I need to know this to decide what `pos.distance` to set for the TMS optimization.

2. Also, my target was inside the brain and was projected to the scalp during the TMS simulation. Is that correct? 
If yes, how do I find the x,y,z and coil orientation coordinates (if computed) from python scripting? I know it can done in the GUI by clicking "Project on Scalp" but I ran my simulations via python on the command line. I looked at a previous discussion (https://github.com/simnibs/simnibs/issues/18) and found this code (as scalp is tissue 5): 
`scalp_surf = head_mesh.crop_mesh(1005)
scalp_pos = scalp_surf.nodes[:]`

But that gives me all the positions and not one value as I would get from the GUI.

3. I also noticed that the TMS optimization gives me the scalp coordinates directly in the final output matrix depending on what value `tmsopt.pos` is set at. (I tried 0mm and 4mm). Is that correct? 

Could you please help?

Thank you in advance.
Best,
Apoorva

**Comment by bjornsig** [comment]
Hi Apoorva,

1. if matsimnibs is provided then pos.distance is not used because the coil position is fully defined by matsiminibs.

2. It's a bit confusing. Can you provide more detail? The projection only happens when you specify a center position, then it happens automatically inside the simulation. However, when you use matsimnibs this is not done. 

3. Yes, that's correct.

**Reply by akelkar26** [reply]
Hi @bjornsig 

Thank you for your response. This clarification was helpful. 

I am sorry point 2 was confusing. 

For my TMS simulations, I got the coordinates from Brainsight by importing the text file and reading it in as mentioned here: https://simnibs.github.io/simnibs/build/html/documentation/neuronavigation/brainsight.html#brainsight

When I look at the elements of the `tms_list.pos[0]`, the center and pos.ydir are `None` and information is in matsimnibs. 

Example here:
`Centre is: None
Pos_ydir is: None
matsimnibs is: [[-6.78800e-01  7.16300e-01 -1.61400e-01  2.04844e+01]
 [ 7.34300e-01  6.62400e-01 -1.48400e-01  1.15479e+01]
 [ 6.00000e-04 -2.19200e-01 -9.75700e-01  5.93516e+01]
 [ 0.00000e+00  0.00000e+00  0.00000e+00  1.00000e+00]]
`
Is it correct to say that pos.distance wasn't used then?

My final goal is to calculate the Euclidian distance between my original TMS simulated target and the target I get from TMS optimization. 

However, the TMS optimization target is on the scalp and my TMS simulated target is in the brain (if pos.distance wasn't used in case matsimnibs was set). 

I would ideally want to get the coordinates of the TMS optimized target inside the brain or the TMS simulated target on the scalp to do a fair Euclidian distance calculation. 

Could you please advise on how to go about this?

Awaiting your reply. 
Thank you
Best,
Apoorva
